traffic_ops/traffic_ops_golang/cdn/capacity.go

```diff
@@ -166,21 +181,39 @@ func addCapacity(cap CapData, cacheStats CacheStats, crStates tc.CRStates, crCon
 		if !strings.HasPrefix(*cache.ServerType, string(tc.CacheTypeEdge)) {
 			continue
 		}
-		if len(stats.KBPS) < 1 || len(stats.MaxKBPS) < 1 {
-			continue
+		var kbps, maxKbps float64
+		for intName, intStats := range stats {
+			if len(intStats.KBPS) < 1 || len(intStats.MaxKBPS) < 1 {
+				continue
+			}
+			if checkServiceInterface(tx, intName, cacheName) {
```

Comment:
Can `checkServiceInterface()` be rewritten so it only needs to be called a single time instead of once per cache?

Reply:
You mean once per interface, or once per cache?

Reply:
Yes I meant instead of once per interface. Ideally it will only be called once overall per request before entering the cache loop. Workflow could be:

1. From your list of cache servers, get all interfaces by server. `ARRAY_AGG` can be used to list more than 1 interface in a single column so that it's only 1 row per server (with no performance penalty)

2. Store the result to a struct

3. In your interface loop, check the interface using that struct as the data source

Reply:
Instead of doing a separate sql query to get the interfaces, we can just make use of the `stats` structure tha we have, where we already have all the interface names. Saves us a DB call. But yeah, I can make it so that the `checkServiceInterface()` function will not need to be called once per interface. Does that make sense?

Reply:
Sounds good to me